### PR TITLE
Modernbert patches

### PR DESCRIPTION
**Pull request by aehrm**
Commits to make ModernBERT and SuperGLEBER QA tests happen:

* [Pass seed to the huggingface trainer in train_hf_qa.py](https://github.com/LSX-UniWue/SuperGLEBer/commit/596ec0a522203f0d0228955e0f50ed5666b1d8a1)
  
  The huggingface Trainer class resets the seed of numpy, torch, etc. to the one
  given in the passed TrainingArguments object, which currently defaults to the
  value given in the huggingface implementation (42).
  
  In order to respect the seed given in the Omegaconf config value (cf.
  `src/train.py`, L32), the config value cfg.seed is passed to the huggingface
  TrainingArguments object, thus allowing to specify custom seed values to the
  training runs.
  
  NB: Since the current default seed provided in `src/conf/config.yaml` is
  identical to that of the huggingface TrainingArguments, this commit will not
  lead to changes relative to previous training runs.
  
* [Add weight initialization to the ModernBERT for QA patch](https://github.com/LSX-UniWue/SuperGLEBer/commit/7ca17e8350f8f4d1b99e201e2f92ce70d9e867c5)
  
  Current (custom patch) implementation of the ModernBertForQuestionAnswering
  model class does not initialize the classifier head, thus resulting in
  unspecified initial values and unstable training runs (incl. nan gradients).
  
  This commit overwrites the weight initialization in the class
  ModernBertForQuestionAnswering by adapting from the current (commit 49b5ab)
  huggingface weight implementation found in the class ModernBertPreTrainedModel.

* [adjust flair/hf training script to take model config args](https://github.com/LSX-UniWue/SuperGLEBer/commit/690a6b8eb234c1500de530219c261340e4fdaa70)
  
  Older GPUs do not have support for torch.compile resp. flash-attention. The
  huggingface implementation ofModernBERT allows to disable these features with
  model config kwargs and `reference_compile=False` and
  `_attn_implementation="sdpa"`, resp.
  
  This commit adds the config key `model_config_args` to the model Omega configs
  to pass such kwargs to the huggingface config.

### CODE REVIEW

File: src/lib_patches/transformers_patches/ModernBERTForQuestionAnswering.py [hunk 1]
```diff
@@ -28,6 +29,37 @@ def __init__(self, config: ModernBertConfig):
 
         self.post_init()
 
+    # this is an adaptation of <https://github.com/huggingface/transformers/blob/49b5ab6a27511de5168c72e83318164f1b4adc43/src/transformers/models/modernbert/modular_modernbert.py#L792>
```
[the remark this answers is not in the document]
Comment (janpf) [reply src/lib_patches/transformers_patches/ModernBERTForQuestionAnswering.py:32]: it's not tho?

File: src/train_flair.py [hunk 2]
```diff
@@ -102,7 +102,8 @@ def training(cfg: DictConfig) -> None:
                 model=cfg.model.model_name,
                 fine_tune=cfg.train_procedure.get("fine_tune", True),
                 force_max_length=True,
-                transformers_model_kwargs=bnb_config | cfg.model.get("model_kwargs", {}),
+                transformers_config_kwargs=cfg.model.get("model_config_args", {}),
```
Comment (janpf) [inline src/train_flair.py:105]: @aehrm is it ok to pass it twice? is this necessary for modernbert?

File: src/lib_patches/transformers_patches/ModernBERTForQuestionAnswering.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (aehrm) [reply src/lib_patches/transformers_patches/ModernBERTForQuestionAnswering.py:32]: it's not

**Review by janpf — APPROVED** [review]
